rss_parser/_parser.py

```diff
@@ -2,42 +2,72 @@
 
 from xmltodict import parse
 
+from rss_parser.custom_decorators import abstract_class_attributes
 from rss_parser.models import XMLBaseModel
+from rss_parser.models.atom import Atom
 from rss_parser.models.rss import RSS
 
 # >>> FUTURE
 # TODO: May be support generator based approach for big rss feeds
 # TODO: Add cli to parse to json
 # TODO: Possibly bundle as deb/rpm/exe
-# TODO: Atom support
-# TODO: Older RSS versions?
+# TODO: Older Atom versions
+# TODO: Older RSS versions
 
 
-class Parser:
-    """Parser for rss files."""
+@abstract_class_attributes("schema")
+class BaseParser:
+    """Parser for rss/atom files."""
 
-    schema: ClassVar[Type[XMLBaseModel]] = RSS
-
-    @staticmethod
-    def _check_atom(root: dict):
-        if "feed" in root:
-            raise NotImplementedError("ATOM feed is not currently supported")
+    schema: ClassVar[Type[XMLBaseModel]]
+    root_key: Optional[str] = None
```

Comment:
If not too much work, I think we can put root_key as ClassVar for schema, so that way it's bound to the schema, as it should be in my opinion, wdyt?

Reply:
I'll think about this a bit more. Not sure how we want to approach it. Honestly, I think I should refactor RSS model a bit so it includes rss child to not use root_key or any of this stuff